### ISSUE TITLE
Warp automatically launches after restart

### ISSUE THREAD

**Issue by shar-mayank** [warpdotdev/Warp#5296]
### Dupe Check

- [X] I have [searched Warp bugs](https://github.com/warpdotdev/warp/issues?q=is%3Aissue+label%3ABugs) and there are no duplicates

### Describe the bug

I'm a Macbook Air M1 2020 user, whenever I restart my laptop it automatically launches Warp even though I haven't set it to "launch at startup" and it is also not showing inside my Login items in my System Settings so there is no way to turn it off.

### To reproduce

I don't know how to reproduce it but you can follow the steps mentioned below:
1. Open Warp and then quit it.
2. Restart your Mac
3. Wait for it to automatically launch when your mac boots up. 

If it doesn't launch automatically in your mac then good for you :)
Note: I'm on beta version of mac right now, but it had happened with me before when I was not on public beta.

### Operating system

MacOS

### Operating system and version

macOS Sequoia Version 15.0 Beta (24A5331b)

### Shell Version

zsh 5.9 (arm64-apple-darwin24.0)

### Current Warp version

v0.2024.09.03.08.02.stable_03

### Regression

No, this bug or issue has existed throughout my experience using Warp

### Does this block you from using Warp daily?

No

### Is this an issue only in Warp?

Yes, I confirmed that this only happens in Warp, not other terminals.

### Warp Internal (ignore): linear-label:b9d78064-c89e-4973-b153-5178a31ee54e

None

**Comment by przemoch-dev** [comment]
Same here. 
MacBook Pro M1 Pro 2021
MacOS 15.1.1

**Comment by marcodallagatta** [comment]
Same behavior here. If I delete the login item it just reappears at the next start of the app. Pretty annoying that this is the default and there's no option to turn it off.

**Comment by alexclst** [comment]
Yeah, I just noticed this, after apparently they added a "feature" of this sort in the most recent update. Why? No app, not Warp, not any app (even of Apple's), should add itself back to the Login Items list after the user has actively removed it. Warp should be updated to do this once, fine. But set a key in its preferences that indicates it has done this, and never do it again. I am also distressed that these re-adding is not causing a macOS notification, which I've seen when manually checking the "launch at login" sort of checkboxes in other apps. It seems Warp is doing this in some very under-the-table way. Even grosser than the basic action.

Now, grant you, macOS itself launches apps that were previously open on login by default, and this behavior effectively ignores the Login Items list. Warp launching that way is fine. But its insistence on being on the Login Items list is ludicrous.

**Comment by frankmak9** [comment]
Same to me, seems like it started after the most recent update. lots of things going wrong with this one my ssh just doesnt feel the same anymore either

**Comment by lucassimines** [comment]
Same here, it's really annoying. I'm thinking about deleting the app

**Comment by gwthompson** [comment]
I just got the update and and realized this was happening. I thought I had missed something but apparently not. It almost makes me want to go back to iTerm! It's so annoying1

**Comment by sensitiky** [comment]
Same here, i remove it from my login items and after every restart appears again

**Comment by jostrasser** [comment]
> Same here, i remove it from my login items and after every restart appears again

Can confirm. Removing it from the login items does not solve the this issue.

**Comment by shpasser** [comment]
Same here. I started experiencing the issue after the last update.

**Comment by derdaani** [comment]
Same here since the latest update (running on v0.2025.03.05.08.02.stable_02). Adding an app to the system's login items without the user's consent is not cool. What makes it worse that there is no option to  revoke this. Manually removing Warp from login items only has an effect until I re-open Warp. This is unacceptable behavior and should be fixed asap.

**Comment by DarsheelCodes** [comment]
If you are running on MacOS Ventura or higher than you have to navigate to **System Settings < General < Login Items & Extensions < Select WARP and click the - (minus) button**
Try restarting and thank me later

**Comment by bryanrieger** [comment]
Removing it from my login items does nothing as it just puts itself back there, and there's no way to turn this behaviour off. Please let us decide which apps we want to launch at login, instead of forcing them down our throats. If it stays like this I'll be ditching Warp within days.

**Comment by angeloevangelista** [comment]
I am not happy with this, but I read somewhere and this worked for me. Marking this "~/Library/Preferences" as "Locked" prevented Warp from adding itself on the Login Itens.

The thing is that I can still add itens on there manually. Well, anxiously awaiting the side effects.

![Image](https://github.com/user-attachments/assets/b654ad05-5b79-4245-8524-f433566dc797)

**Comment by haris-maverick1** [comment]
same for me after recent update. making me lose faith in warp.

**Comment by danielefranceschi** [comment]
Also #6051

**Comment by emirchus** [comment]
Same here since v0.2025.03.05.08.02.stable_02

**Comment by alexclst** [comment]
Still the case with v0.2025.03.12.08.02.stable_03.

**Comment by Prochy20** [comment]
Same here with `v0.2025.03.12.08.02.stable_03`

**Comment by jostrasser** [comment]
@dannyneira may we get an update on this? Thanks!

**Comment by dannyneira** [comment]
Hey folks, In the [latest version](https://docs.warp.dev/getting-started/changelog#id-2025.03.12-v0.2025.03.12.08.02), there is now and option in the settings for having Warp open on startup (login item). Note if you have the Warp setting toggled on, and you right-click on Warp's app icon and uncheck "Open at login", Warp will still be added as a login item upon restart. You need to toggle this in Warp's setting for it to persist.

Control whether Warp starts at login via a setting under Settings > Features > Start Warp at login (MacOS only).

Please update to the latest and let us know if any issues.

**Comment by jostrasser** [comment]
> Hey folks, In the [latest version](https://docs.warp.dev/getting-started/changelog#id-2025.03.12-v0.2025.03.12.08.02), there is now and option in the settings for having Warp open on startup (login item). Note if you have the Warp setting toggled on, and you right-click on Warp's app icon and uncheck "Open at login", Warp will still be added as a login item upon restart. You need to toggle this in Warp's setting for it to persist.

Hey @dannyneira 
Thanks for the fast feedback! :)

I tested it now and it works like a charm :)

![Image](https://github.com/user-attachments/assets/bb7aa5a9-b82c-41a1-b6f4-a90918995def)

I think many of us have overlooked this switch ;)

Thanks for your support!

BR/JO!

**Comment by shar-mayank** [comment]
> Hey folks and [@shar-mayank](https://github.com/shar-mayank) , In the [latest version](https://docs.warp.dev/getting-started/changelog#id-2025.03.12-v0.2025.03.12.08.02), there is now and option in the settings for having Warp open on startup (login item). Note if you have the Warp setting toggled on, and you right-click on Warp's app icon and uncheck "Open at login", Warp will still be added as a login item upon restart. You need to toggle this in Warp's setting for it to persist.
> 
> Control whether Warp starts at login via a setting under Settings > Features > Start Warp at login (MacOS only).
> 
> Please update to the latest and let us know if any issues.

Thank you so much for this, I tested it out and it works perfectly.
